### PR TITLE
linuxptp: improvements

### PR DESCRIPTION
**Pull request by pberginkonsult**
A couple of improvements to the linuxptp recipe. We use the install function from the makefile in order to install all produced binaries and manual pages. Also ship and package example configurations in a separate package. Inspiration is taken from how debian package linuxptp.

### CODE REVIEW

File: meta-oe/recipes-connectivity/linuxptp/linuxptp_3.1.bb [hunk 1]
```diff
@@ -10,14 +10,12 @@ SRC_URI = "http://sourceforge.net/projects/linuxptp/files/v${PV}/linuxptp-${PV}.
 SRC_URI[md5sum] = "2264cb69c9af947028835c12c89a7572"
 SRC_URI[sha256sum] = "f58f5b11cf14dc7c4f7c9efdfb27190e43d02cf20c3525f6639edac10528ce7d"
 
-EXTRA_OEMAKE = "ARCH=${TARGET_ARCH} EXTRA_CFLAGS='${CFLAGS}'"
+EXTRA_OEMAKE = "ARCH=${TARGET_ARCH} EXTRA_CFLAGS='${CFLAGS}' mandir=${mandir}"
```
Comment (kraj) [inline meta-oe/recipes-connectivity/linuxptp/linuxptp_3.1.bb:14]: should this `inherit manpages` perhaps ?

File: meta-oe/recipes-connectivity/linuxptp/linuxptp_3.1.bb [hunk 2]
```diff
@@ -17,6 +17,13 @@ export KBUILD_OUTPUT="${RECIPE_SYSROOT}"
 
 do_install() {
     oe_runmake install DESTDIR=${D} prefix=${prefix}
+
+    # Install example configs from source tree
+    install -d ${D}${docdir}/${PN}
+    cp -r ${S}/configs ${D}${docdir}/${PN}
```
Comment (kraj) [inline meta-oe/recipes-connectivity/linuxptp/linuxptp_3.1.bb:23]: I wonder if using something like below should be used
```
cp -R --no-dereference --preserve=mode,links
```

**Review by kraj — CHANGES_REQUESTED** [review]

File: meta-oe/recipes-connectivity/linuxptp/linuxptp_3.1.bb (hunk 1 above)
Comment (pberginkonsult) [reply meta-oe/recipes-connectivity/linuxptp/linuxptp_3.1.bb:14]: The man pages in linuxptp is not built, they are in the repo from the beginning. There is neither a configuration option to enable/disable the installation of them. (https://salsa.debian.org/multimedia-team/linuxptp/-/blob/master/makefile#L86) For that reason I don't see any need for 'inherit manpages' that create the PACKAGECONFIG manpages.

Then I see in manpages.bbclass that there is some indexing in pkg_postinst (http://git.yoctoproject.org/cgit/cgit.cgi/poky/tree/meta/classes/manpages.bbclass#n15) that probably coudl be useful for linuxptp.

So I'm a bit unsure, please advise if you have an opinion. We can probably make the assumption that inherit manpages does not harm anything.

File: meta-oe/recipes-connectivity/linuxptp/linuxptp_3.1.bb (hunk 2 above)
Comment (pberginkonsult) [reply meta-oe/recipes-connectivity/linuxptp/linuxptp_3.1.bb:23]: I have looked in oe-core and meta-openembedded and can not find a common pattern around this. But agree that your suggestion is better if in the future links show up among the installed files.

**Comment by kraj** [comment]
thanks I have staged it in `master-next`

**Review by kraj — APPROVED** [review]
LGTM

**Comment by kraj** [comment]
this is in master now.

**Pull request closed without merging**